storage/innobase/include/btr0btr.h

```diff
@@ -83,18 +83,19 @@ btr_root_adjust_on_import(
 ATTRIBUTE_COLD void btr_decryption_failed(const dict_index_t &index);
 
 /** Get an index page and declare its latching order level.
-@param[in]	index	index tree
-@param[in]	page	page number
-@param[in]	mode	latch mode
-@param[in]	merge	whether change buffer merge should be attempted
-@param[in,out]	mtr	mini-transaction
-@param[out]	err	error code
-@param[out]	first	set if this is a first-time access to the page
+@param	index	index tree
+@param	page	page number
+@param	latch_mode	latch mode
+@param	merge	whether change buffer merge should be attempted
+@param	mtr	mini-transaction
+@param	err	error code
+@param	first	set if this is a first-time access to the page
+@param		page_get_mode	page get mode
 @return block */
-buf_block_t *btr_block_get(const dict_index_t &index,
-                           uint32_t page, rw_lock_type_t mode, bool merge,
-                           mtr_t *mtr, dberr_t *err= nullptr,
-                           bool *first= nullptr);
+buf_block_t *btr_block_get(const dict_index_t &index, uint32_t page,
+                           rw_lock_type_t latch_mode, bool merge, mtr_t *mtr,
+                           dberr_t *err= nullptr, bool *first= nullptr,
+                           ulint page_get_mode= BUF_GET);
```

Comment:
The parameter `mode` was renamed to `latch_mode`, but not in the function comment in `btr0btr.cc`. In this file, the formatting of the `@param` comments got uglier beacuse one of the parameter names is longer. Could we retain the old naming?

Instead of changing the function signature, could we add a debug-only field to `mtr_t` to denote that `BUF_GET` should be treated as `BUF_GET_POSSIBLY_FREED`? That would avoid a potential performance regression while keeping the debug check happy. After all, `BUF_GET` and `BUF_GET_POSSIBLY_FREED` are equivalent in non-debug builds.